src/data/fin.lean

```diff
@@ -488,6 +488,36 @@ lemma succ_above_lt_gt (p : fin (n + 1)) (i : fin n) : i.cast_succ < p ∨ p < i
 or.cases_on (succ_above_lt_ge p i)
   (λ h, or.inl h) (λ h, or.inr (lt_of_le_of_lt h (cast_succ_lt_succ i)))
 
+/-- Embedding `i : fin n` into `fin (n + 1)` using a pivot `p` that is greater
+results in a value that is less than `p`. -/
+lemma succ_above_lt_iff (p : fin (n + 1)) (i : fin n) : i.cast_succ < p ↔ p.succ_above i < p :=
```

Comment:
My only query is the naming. With this lemma name, I would have expected the flip of this `iff`.

Reply:
In fact, flipping this lemma probably makes it a good `simp` lemma!